File: apps/android/app/src/main/java/com/barns/app/presentation/myitems/ItemOfficialContent.kt
package com.barns.app.presentation.myitems

/**
 * Concise, presentation-only view of the official read-only content shown as
 * supporting after-care reference on a registered greenery's detail screen.
 *
 * Boundary: this is built from official [com.barns.app.domain.model.GreeneryInfo]
 * / [com.barns.app.domain.model.CareGuide] content only. It carries no
 * customer-owned [com.barns.app.domain.model.ProductItem] state and is never
 * submitted anywhere — it is reference information, not personalized data.
 */
data class ItemOfficialContent(
    val overview: String?,
    val lightPreference: String?,
    val wateringOverview: String?,
    val careGuides: List<CareGuideSummary>,
) {
    data class CareGuideSummary(
        val id: String,
        val title: String,
        val summary: String,
    )

    val hasBasicInformation: Boolean
        get() = overview != null || lightPreference != null || wateringOverview != null

    val hasCareGuides: Boolean
        get() = careGuides.isNotEmpty()

    val isEmpty: Boolean
        get() = !hasBasicInformation && !hasCareGuides
}

File: apps/ios/Barns/BarnsApp/Presentation/MyItems/ItemOfficialContent.swift
/// Concise, presentation-only view of the official read-only content shown as
/// supporting after-care reference on a registered greenery's detail screen.
///
/// Boundary: this is built from official `GreeneryInfo` / `CareGuide` content
/// only. It carries no customer-owned `ProductItem` state and is never
/// submitted anywhere — it is reference information, not personalized data.
struct ItemOfficialContent: Equatable {
    struct CareGuideSummary: Equatable, Identifiable {
        let id: String
        let title: String
        let summary: String
    }

    /// Official basic information (from `GreeneryInfo`), when available.
    let overview: String?
    let lightPreference: String?
    let wateringOverview: String?
    /// A short list of official care-guide summaries (kept to a few rows).
    let careGuides: [CareGuideSummary]

    var hasBasicInformation: Bool {
        overview != nil || lightPreference != nil || wateringOverview != nil
    }

    var hasCareGuides: Bool { !careGuides.isEmpty }

    var isEmpty: Bool { !hasBasicInformation && !hasCareGuides }
}
